src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -838,7 +838,11 @@ public virtual int Run(TextWriter consoleOutput, CancellationToken cancellationT
             driver = driver.RunGeneratorsAndUpdateCompilation(input, out var compilationOut, out var diagnostics);
             generatorDiagnostics.AddRange(diagnostics);
 
-            if (!disableCache)
+            // We only cache the generator driver if it produced any generated files. While its possible that it was expensive
+            // to calculate that nothing needed to be generated, real world usage has found that generators are generally only
+            // expensive when actually producing source. By only caching those with results, we help to keep memory usage down
+            // when it probably wouldn't improve the performance anyway.
+            if (!disableCache && driver.GetRunResult().GeneratedTrees.Any())
```

Comment:
I'm curious about this - caching empty results increases memory usage? Shouldn't empty results be small?

Reply:
What is the ballpark for how much the cache costs in terms of memory usage?

Reply:
The driver holds onto copies of things like the Compilation, as well as all the intermediate nodes + state tables. So even if it doesn't produce anything, with enough files it can be a non-trivial amount of state.